### ISSUE TITLE
Poco::Data - Sqlite bind STL and a primitive data type at the same time

### ISSUE THREAD

**Issue by sawyna** [pocoproject/poco#2893]
I'm using POCO libraries to connect to a sqlite DB.
I'm trying to pass in a list of strings to one column and an integer to another column in a single update statement. For some reason binding the variables does not work. I have tried multiple approaches. 

- Pass values separately for each column:
```
vector<string> listOfValues = {"VAL_STR1", "VAL_STR2"};
int intVal = 1234;
session << "UPDATE SomeTable SET Column1=<SomeVal> WHERE ListColumn = ? AND IntegerColumn=?", bind(listOfValues), bind(intVal), now;
```

Error: `Exception: Binding error: Size mismatch in Bindings. All Bindings MUST have the same size`

- Pass as a tuple
```
vector<string> listOfValues = {"VAL_STR1", "VAL_STR2"};
int intVal = 1234;
typedef Poco::Tuple<vector<string>, int> ParamTuple;
ParamTuple paramTuple(listOfValues, intVal);
session << "UPDATE SomeTable SET Column1=<SomeVal> WHERE ListColumn = ? AND IntegerColumn=?",  bind(paramTuple), now;
```
Error: `Exception: Not implemented: std::vector binder must be implemented.`

Things I observed:
- The first approach seems to work with multiple binding values when the vector size is 1
- The first approach works well for any vector size when there's only vector in the bind params and nothing else.

I suppose this is a common use wanting to pass a list of values and a single value at the same time or am I missing something?

**Comment by aleks-f** [comment]
Obviously, that is not a "common use"; as the errors indicate, bindings must have the same number of elements - both scalars, or both containers of same size. If you want to use tuple, then you need a vector of tuples, not the other way around.

**Comment by sawyna** [comment]
> Obviously, that is not a "common use";

It's fairly common and basic to write a query `SELECT * FROM Table WHERE Column1 = <something> AND Column2 in <a list of something>`. Why would that not be a **common** use case? 

> as the errors indicate, bindings must have the same number of elements - both scalars, or both containers of same size

For me this feels like a restriction in terms of library usage. This means I cannot write queries if the where clause has varying number of values for each column.

How would you suggest that I write the query mentioned in the issue using the library?

**Comment by aleks-f** [comment]
Well, it depends whether you consider things common for SQL or Data library - I referred to common as in "common for Data library as is now".  Furthermore, `in` is a different thing from `=`, and is not supported via bindings (I'd also recommend using `use` instead of `bind`). If you want to use `=` in SQL `where` clause, then obviously you will have to pass in a scalar value, even in SQL. If you want to bind in insert or update portion of SQL, then you can use scalars or STL containers (which all must be of the same size). If you want to use `in`, you can assemble your own SQL string containing the list. And finally, if you don't like the way things are now in an open source library, you are welcome to attempt to make it as you think it should be and send a pull request.

**Comment by sawyna** [comment]
>  If you want to use in, you can assemble your own SQL string containing the list

Thank you. That's what I need. I'm already using it this way and wanted to know if there's an alternative if I'm missing. 

> And finally, if you don't like the way things are now in an open source library, you are welcome to attempt to make it as you think it should be and send a pull request.

Apologies if it has been communicated wrongly. The information I requested is not mentioned in the documentation anywhere. It's nowhere stated explicitly that this library doesn't support a combination of `=` and `in` queries. Hence I raised the issue whether it is supported.
